framework/include/libtorch/utils/LibtorchArtificialNeuralNet.h

```diff
@@ -58,8 +58,9 @@ class LibtorchArtificialNeuralNet : public torch::nn::Module, public LibtorchNeu
    * Overriding the forward substitution function for the neural network, unfortunately
    * this cannot be const since it creates a graph in the background
    * @param x Input tensor for the evaluation
+   * @param classify Flag for classification mode, defaults to regression mode if false
    */
-  virtual torch::Tensor forward(torch::Tensor & x) override;
+  virtual torch::Tensor forward(torch::Tensor & x, bool classify) override;
```

Comment:
I think the right approach here would be to add an output activation to the construction of the NNs. If it is left empty then nothing otherwise the activation function the user describes. I would like to keep the forward function clean to mimic the original libtorch modules. I am little confused why we don't get compilation warning this does not override a function.

Reply:
I felt the same with regards to the forward function. I will try your idea and submit a commit.